spec/models/miq_product_feature_spec.rb

```diff
@@ -16,19 +14,48 @@
     )
   end
 
+  def traverse_product_features(product_feature, &block)
+    block.call(product_feature)
+    if product_feature.key?(:children)
+      product_feature[:children].each { |child| traverse_product_features(child, &block) }
+    end
+  end
+
+  RSpec::Matchers.define :have_product_feature_structure do
```

Comment:
I can create more tests ;)

Yeah, honestly the matcher wasn't the first thing I jumped to. I'd be OK with just a method but it seemed like I'd have to stuff a lot of logic into that method. Open to suggestions :)